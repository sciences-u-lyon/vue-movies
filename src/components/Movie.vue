<template>
  <article class="media">
    <div class="columns is-variable">
      <div class="column is-one-quarter">
        <img :src="poster">
      </div>
      <div class="column">
        <div class="media-content">
          <div class="content">
            <strong>{{ title }}</strong>
            <br>
            <small>{{ subtitle }}</small>
            <p class="abstract">{{ abstract }}</p>

            <slot></slot>
          </div>
        </div>
      </div>
    </div>
  </article>
</template>

<script>
export default {
  props: {
    poster: String,
    title: String,
    subtitle: String,
    abstract: String
  }
}
</script>
