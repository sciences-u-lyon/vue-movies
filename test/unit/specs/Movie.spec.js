import { mount } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Movie from '@/components/Movie.vue'

describe('Movie.vue', () => {
  it('should render Movie with correct contents', () => {
    const wrapper = mount(Movie, {
      propsData: {
        poster: '/poster.jpg',
        title: 'A movie title',
        subtitle: 'A movie subtitle',
        abstract: 'Lorem Ipsum'
      },
      slots: {
        default: '<button>14:00</button>'
      }
    })
    const vm = wrapper.vm

    const renderer = createRenderer()
    renderer.renderToString(vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
