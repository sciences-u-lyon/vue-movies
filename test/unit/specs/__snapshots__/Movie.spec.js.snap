// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie.vue should render Movie with correct contents 1`] = `
<article class="media">
    <figure class="media-left">
        <img src="/poster.jpg">
    </figure>
    <div class="media-content">
        <div class="content"><strong>A movie title</strong>
            <br>
            <small>A movie subtitle</small>
            <p class="abstract">Lorem Ipsum</p>
            <button>14:00</button>
        </div>
    </div>
</article>
`;
